src/service/viewport/viewport-binding-ios-embed-wrapper.js

```diff
@@ -37,11 +39,14 @@ const TAG_ = 'Viewport';
 export class ViewportBindingIosEmbedWrapper_ {
 
   /**
-   * @param {!Window} win
+   * @param {!../ampdoc-impl.AmpDoc} ampdoc
```

Comment:
IMPORTANT: This might be a problem. I believe the viewport bindings are by nature global - window-based. This may break PWA code. `Viewport` class is meant as an adapter from a global viewport binding to the ampdoc. Can we rephrase via `Viewport` class?

Reply:
> Can we rephrase via Viewport class?

I don't follow.

Reply:
Currently, the structure is this:

- ViewportBinding is roughly a wrapper around global window. Thus it's also global.
- Viewport service is an ampdoc service. I.e. there are as many Viewport instances as ampdocs. As a matter of fact, that's the purpose of Viewport service - it's an adopter between an ampdoc and a global window (ViewportBinding)
- Ampdocs and window are N:1. I.e. multiple ampdocs can share the same window.

So, ViewportBinding has to remain global window scoped. We can't use ampdoc in the constructor or really anywhere in that class. This change, if deployed as currently written, will break the existing PWAs.